src/components/ErrorDialog/index.js

```diff
@@ -10,67 +10,68 @@ import { FormattedMessage, injectIntl } from 'react-intl';
 import styles from './styles';
 import appActions from '../../redux/App/actions';
 import topicActions from '../../redux/Topic/actions';
+import graphqlActions from '../../redux/Graphql/actions';
 
 @injectIntl
 @withStyles(styles, { withTheme: true })
 @connect((state) => ({
-  errorApp: state.App.get('errorApp'),
-  errorTopic: state.Topic.get('errorTopic'),
-}), (dispatch) => ({
-  clearErrorApp: () => dispatch(appActions.clearErrorApp()),
-  clearErrorTopic: () => dispatch(topicActions.clearErrorTopic()),
+  get error() {
+    let type = '';
+    let error = { message: '', route: '' };
+    if (state.App.get('errorApp')) {
+      type = 'app';
+      error = state.App.get('errorApp');
+    } else if (state.Topic.get('errorTopic')) {
+      type = 'topic';
+      error = state.Topic.get('errorTopic');
+    } else if (state.Graphql.get('error')) {
+      type = 'graphql';
+      error = state.Graphql.get('error');
+    }
+    return { ...error, type };
+  },
 }))
 export default class ErrorDialog extends Component {
   static propTypes = {
     classes: PropTypes.object.isRequired,
-    errorApp: PropTypes.object,
-    clearErrorApp: PropTypes.func.isRequired,
-    errorTopic: PropTypes.object,
-    clearErrorTopic: PropTypes.func.isRequired,
+    error: PropTypes.object,
+    dispatch: PropTypes.func.isRequired,
   }
 
   static defaultProps = {
-    errorApp: undefined,
-    errorTopic: undefined,
+    error: {},
   }
 
-  render() {
-    const {
-      classes,
-      errorApp,
-      errorTopic,
-    } = this.props;
-
-    const isOpen = Boolean(errorApp || errorTopic);
-    let error;
-    if (errorApp) {
-      error = errorApp;
-    } else if (errorTopic) {
-      error = errorTopic;
+  clearError = () => {
+    const { error, dispatch } = this.props;
+    if (!error.type) {
+      console.error('No error to clear. Check <ErrorDialog />'); // eslint-disable-line
+      return;
     }
+    const clearError = {
+      app: appActions.clearErrorApp,
+      topic: topicActions.clearErrorTopic,
+      graphql: graphqlActions.clearGraphqlError,
+    }[error.type];
+    dispatch(clearError());
+  }
 
-    if (!error) {
-      return null;
-    }
+  render() {
+    const { classes, error } = this.props;
 
     return (
-      <Dialog open={isOpen}>
+      <Dialog open={!!error.message}>
         <DialogTitle><FormattedMessage id="str.error" defaultMessage="Error" /></DialogTitle>
         <DialogContent>
           <Typography className={classes.errorRoute}>{error.route}</Typography>
```

Comment:
Actually, I'm glad you commented on this. Do we need the route? Not sure exactly what the use case is here?

Reply:
This is more for dev purposes. Since the UI errors are only visible to the user. It's easy for the users to tell us which route gave them an error. Until we have remote logging in which we can see an error on the UI side I'd like to keep this in there.